### ISSUE TITLE
Remove demo for checkboxes within mdc-list, until we can figure out the a11y requirements

### ISSUE THREAD

**Issue by lynnjepsen** [material-components/material-components-web#2849]
We had made some demos of mdc-list with checkboxes, but the keyboard navigation of this demo is unclear. We believe all of our components should have a clear focus order:
https://material.io/design/usability/accessibility.html#composition

So we're going to deprecate support for checkboxes until we can figure out the right way to navigate a multi-select list with a keyboard.

**Comment by kfranqueiro** [comment]
These have since been removed (#2883) and re-added after implementing the List component with keyboard support.
